core/src/replay_stage.rs

```diff
@@ -534,61 +512,46 @@ impl ReplayStage {
         bank_progress: &mut ForkProgress,
         transaction_status_sender: Option<TransactionStatusSender>,
         verify_recyclers: &VerifyRecyclers,
-    ) -> (result::Result<(), BlocktreeProcessorError>, usize) {
-        let mut tx_count = 0;
-        let now = Instant::now();
-        let load_result =
-            Self::load_blocktree_entries_with_shred_info(bank, blocktree, bank_progress)
-                .map_err(BlocktreeProcessorError::FailedToLoadEntries);
-        let fetch_entries_elapsed = now.elapsed().as_micros();
-        if load_result.is_err() {
-            bank_progress.stats.fetch_entries_fail_elapsed += fetch_entries_elapsed as u64;
-        } else {
-            bank_progress.stats.fetch_entries_elapsed += fetch_entries_elapsed as u64;
-        }
+    ) -> result::Result<usize, BlocktreeProcessorError> {
+        let tx_count_before = bank_progress.replay_progress.num_txs;
+        let confirm_result = blocktree_processor::confirm_slot(
+            blocktree,
+            bank,
+            &mut bank_progress.replay_stats,
+            &mut bank_progress.replay_progress,
+            false,
+            transaction_status_sender,
+            None,
+            verify_recyclers,
+        );
+        let tx_count_after = bank_progress.replay_progress.num_txs;
+        let tx_count = tx_count_after - tx_count_before;
 
-        let replay_result = load_result.and_then(|(entries, num_shreds, slot_full)| {
-            trace!(
-                "Fetch entries for slot {}, {:?} entries, num shreds {}, slot_full: {}",
-                bank.slot(),
-                entries.len(),
-                num_shreds,
-                slot_full,
-            );
-            tx_count += entries.iter().map(|e| e.transactions.len()).sum::<usize>();
-            Self::replay_entries_into_bank(
-                bank,
-                bank_progress,
-                entries,
-                num_shreds,
-                slot_full,
-                transaction_status_sender,
-                verify_recyclers,
-            )
-        });
+        confirm_result.map_err(|err| {
+            let slot = bank.slot();
+            warn!("Fatal replay error in slot: {}, err: {:?}", slot, err);
+            if let BlocktreeProcessorError::InvalidBlock(_) = &err {
+                let last_entry = &bank_progress.replay_progress.last_entry;
+                datapoint_error!(
```

Comment:
Should we also submit datapoints for the other types of errors? I think `BlocktreeProcessorError::InvalidBlock` is only for failures from `verify_ticks` right?

Reply:
Oh nvm, I see the `replay-stage-mark_dead_slot` metric below, is there a reason we have a specific metric just for `replay-stage-block-error`?